contracts/BaseWrapper.sol

```diff
@@ -107,19 +107,31 @@ abstract contract BaseWrapper {
             token.safeApprove(address(_bestVault), uint256(-1)); // Vaults are trusted
         }
 
-        // `receiver` now has shares of `_bestVault` (worth `amount` tokens) as balance
-        return
-            _bestVault
-                .deposit(amount, receiver)
-                .mul(_bestVault.pricePerShare()) // Adjust by price of best
-                .div(10**uint256(_bestVault.decimals()));
+        // Depositing returns number of shares deposited
+        // NOTE: Shortcut here is assuming the number of tokens deposited is equal to the
+        //       number of shares credited, which helps avoid an occasional multiplication
+        //       overflow if trying to adjust the number of shares by the share price.
+        uint256 beforeBal = token.balanceOf(address(this));
+        if (receiver != address(this)) {
+            _bestVault.deposit(amount, receiver);
+        } else if (amount < uint256(-1)) {
```

Comment:
can also use from solidity 0.6.8 and above as a constant this value and reuse it across the file
```
    // 5) Type (Solidity 0.6.8)
    uint256 constant public MAX_INT_TYPE = type(uint256).max;
 ```